### PR TITLE
add controller-user branch for user

### PR DESCRIPTION
**Pull request by HammamAbuShehadeh**
#13 add user controller

### CODE REVIEW

**Comment by HammamAbuShehadeh** [comment]
Done Update scheme models files

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: i will start review

File: server/server/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const bcrypt = require("bcrypt");
+const cookie = require("cookie");
+
+const BlackListModel = require("../models/blackList");
+const UserModel = require("../models/user");
+const DishModel = require("../models/dish");
+// const CartModel = require("../models/cart");
+const OrderModel = require("../models/order");
+const RestaurantModel = require("../models/restaurant");
+
```
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:10]: nice import.

**Review by AHMADJUAM — COMMENTED** [review]
looks good.

File: server/server/controllers/userController.js [hunk 2]
```diff
@@ -25,6 +25,23 @@ const getPastOrders = async (req, res) => {
   }
 };
 
+// Get current active order ok
+const getCurrentOrders = async (req, res) => {
+  try {
+    const { userID } = req.query; //add New line
```
Comment (AHMADJUAM) [inline server/server/controllers/userController.js]: please remove the comment.

**Review by AHMADJUAM — COMMENTED** [review]
the remining code looks good.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: thank you farah for editing the code and adding new function,i will check them now.

File: server/server/controllers/userController.js (hunk 2 above)
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js]: Done

File: server/server/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+const bcrypt = require("bcrypt");
+const cookie = require("cookie");
+
+const BlackListModel = require("../models/blackList");
+const UserModel = require("../models/user");
+const DishModel = require("../models/dish");
+// const CartModel = require("../models/cart");
```
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:7]: code looks clean but why this line is in comment if we dont need it in our code please remove it.
with all respect Ahmad Juma

**Review by AHMADJUAM — COMMENTED** [review]
the code looks fine please cheack the review comments i righte.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: best regards for farah.
thanks farah for adding get all dishes and filter it by category .
i'll start reviewing the code now.

**Review by AHMADJUAM — COMMENTED** [review]
the code seems clean thanks for your hard working.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: thanks for adding search for dish  function.
i will reviw itfor you now.

File: server/server/controllers/userController.js [hunk 4]
```diff
@@ -89,10 +89,42 @@ const getDishItems = async (req, res) => {
   }
 };
 
+// search for Dish ok
+const searchDishes = async (req, res) => {
+  try {
+    const { minPrice, maxPrice, category, name } = req.query;
+
+    // Define the search criteria
+    const searchCriteria = {};;
```
Comment (AHMADJUAM) [inline server/server/controllers/userController.js]: please remove the extra semicolon in line 98.

**Review by AHMADJUAM — COMMENTED** [review]
your code is fine just quick review to my comment.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: thanks for providing get single Dish information  function.

File: server/server/controllers/userController.js (hunk 4 above)
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js]: Done

**Review by AHMADJUAM — COMMENTED** [review]
the code looks clean thanks for adding get single Dish information function.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: السلام عليكم ورحمة الله وبركاته.
سأبدأ ان شاء الله بتزكية هذا الكود بعون الله.

**Review by AHMADJUAM — COMMENTED** [review]
هذا الكود يبدوا سليما جزاك الله خيرا يا فرح.

والسلام عليكم ورحمتو الله وبركاته.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: yooo
i'll satrt cheacking your code YOOO.
be ready YOO.

**Review by AHMADJUAM — COMMENTED** [review]
the code looks fine yoo.
thank you yoo.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: hey i will cheack the code for you.

**Review by AHMADJUAM — COMMENTED** [review]
the code is clean thanks for your work.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: hey farah i will check your code

**Review by AHMADJUAM — COMMENTED** [review]
the code is good excelent  work.

File: server/server/controllers/userController.js
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js:1]: thanks

File: server/server/controllers/userController.js (hunk 4 above)
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js]: Done

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: i will cheack the function to update user data and function to get users Data now.

**Review by AHMADJUAM — COMMENTED** [review]
the code fine thank you.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: hey there i will cheack the code now.

**Review by AHMADJUAM — COMMENTED** [review]
clean code thank you.

File: server/server/controllers/userController.js
Comment (AHMADJUAM) [inline server/server/controllers/userController.js:1]: i will cheack the code for you now

File: server/server/controllers/userController.js [hunk 5]
```diff
@@ -410,6 +410,41 @@ const getTokens = async (_, res) => {
...
+    } else {
+      res
+        .status(422)
+        .json({ message: "The user you are trying to delete wasn't found" });
+    }
+  } catch (error) {
+    res.status(500).json(error.message);
+  }
+};
+//new function
```
Comment (AHMADJUAM) [inline server/server/controllers/userController.js]: please remove the comment its unnecessary.

**Review by AHMADJUAM — COMMENTED** [review]
looks good please cheack my reviews.
best regards Ahmad Juma.

File: server/server/controllers/userController.js (hunk 3 above)
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js:7]: I will check

File: server/server/controllers/userController.js (hunk 5 above)
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js]: Done

File: server/server/controllers/userController.js
Comment (FarahAlsoqi) [reply server/server/controllers/userController.js:1]: thanks

**Pull request merged**